iris/src/iris/pipeline/delete_lecture_units_pipeline.py

```
@@ -64,24 +64,26 @@ def __init__(
-                self.delete_lecture_unit(lecture_unit)
-            self.callback.done("Lecture unit removed")
+                if not self.delete_page_chunk(lecture_unit):
+                    return False
+                if not self.delete_transcriptions(lecture_unit):
+                    return False
+                if not self.delete_lecture_unit_segments(lecture_unit):
+                    return False
+                if not self.delete_lecture_unit(lecture_unit):
+                    return False
+            return True
         except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)